core/nixpkgs.bzl

```diff
@@ -106,18 +106,40 @@ def nixpkgs_git_repository(
     )
 
 def _nixpkgs_local_repository_impl(repository_ctx):
-    if not bool(repository_ctx.attr.nix_file) != \
-       bool(repository_ctx.attr.nix_file_content):
-        fail("Specify one of 'nix_file' or 'nix_file_content' (but not both).")
+    if bool(repository_ctx.attr.nix_file) and bool(repository_ctx.attr.nix_file_content) or \
+       bool(repository_ctx.attr.nix_file) and bool(repository_ctx.attr.nix_flake_lock_file) or \
+       bool(repository_ctx.attr.nix_flake_lock_file) and bool(repository_ctx.attr.nix_file_content):
+        fail("Specify only one of 'nix_file', 'nix_file_content' or 'nix_flake_lock_file'.")
+
     if repository_ctx.attr.nix_file_content:
         target = "default.nix"
         repository_ctx.file(
             target,
             content = repository_ctx.attr.nix_file_content,
             executable = False,
         )
-    else:
+    elif repository_ctx.attr.nix_file:
         target = cp(repository_ctx, repository_ctx.attr.nix_file)
+    else:
+        lock_filename = str(repository_ctx.attr.nix_flake_lock_file).split(":")[1]
```

Comment:
Maybe there is a better way to get the filename?

Reply:
You can follow the `cp` pattern above, I think.
```
lock_filename = cp(repository_ctx, repository_ctx.attr.nix_flake_lock_file).
```